src/fides/api/db/system.py

```diff
@@ -147,57 +150,73 @@ async def upsert_privacy_declarations(
 async def upsert_cookies(
     async_session: AsyncSession,
     cookies: Optional[List[Dict]],
-    privacy_declaration: PrivacyDeclaration,
-    system: System,
+    privacy_declaration: Optional[PrivacyDeclaration],
+    system: Optional[System],
 ) -> None:
     """
-    Upsert cookies for the given privacy declaration: retrieve cookies
-    by name/system/privacy declaration.
+    Upsert cookies for the given system or privacy_declaration.  Cookies can be attached at the system-level
+    or the privacy declaration-level.
 
     Remove any existing cookies that aren't specified here.
     """
+    if not (privacy_declaration or system):
+        # Dev-level error
+        raise Exception(
```

Comment:
nit: this may actually be a good use case for `ValueError` :) ?
```suggestion
        raise ValueError(
```